server/bleep/src/indexes/file.rs

```diff
@@ -537,109 +539,135 @@ impl File {
         }
         trace!("added cache entry");
 
-        let lang_str = if file.kind.is_file() {
-            repo_metadata
-                .langs
-                .get(&entry_pathbuf, file.buffer.as_ref())
-                .unwrap_or_else(|| {
-                    warn!(?entry_pathbuf, "Path not found in language map");
-                    ""
-                })
-        } else {
-            ""
-        };
+        let last_commit = repo_metadata.last_commit_unix_secs;
 
-        // calculate symbol locations
-        let symbol_locations = if file.kind.is_file() {
-            // build a syntax aware representation of the file
-            let scope_graph = TreeSitterFile::try_build(file.buffer.as_bytes(), lang_str)
-                .and_then(TreeSitterFile::scope_graph);
-
-            match scope_graph {
-                // we have a graph, use that
-                Ok(graph) => SymbolLocations::TreeSitter(graph),
-                // no graph, it's empty
-                Err(err) => {
-                    warn!(?err, %lang_str, "failed to build scope graph");
-                    SymbolLocations::Empty
-                }
+        match dir_entry {
+            RepoDirEntry::Dir(dir) => {
```

Comment:
Can you extract these into separate methods on `RepoDirEntry` for each enum variant?
It would possibly help our testing story to verify the logic dependencies here outside of this massive function body.

Maybe rust-analyzer's extract function is a good start separating these out.